engine/tests/phpunit/Elgg/UserCapabilitiesTest.php

```diff
@@ -0,0 +1,512 @@
+			'owner_guid' => 24,
+		]);
+		$metadata = new \ElggMetadata((object) [
+					'owner_guid' => 0,
+					'entity_guid' => $entity->guid,
+		]);
+
+		$this->setupServices();
+		$this->assertTrue($entity->canEditMetadata($metadata, 24));
+	}
+
+	/**
+	 * @group UserCapabilities
+	 */
+	public function testOwnerCanEditMetadataOwnedBySelf() {
+		$this->setupEntity(26, 'user', '');
+		$this->setupEntity(27, 'user', '');
+		$entity = $this->setupEntity(28, 'object', 'foo', [
+			'owner_guid' => 26,
+		]);
+		$metadata = new \ElggMetadata((object) [
+					'owner_guid' => 26,
+					'entity_guid' => $entity->guid,
+		]);
+
+		$this->setupServices();
+		$this->assertTrue($entity->canEditMetadata($metadata, 26));
+		$this->assertFalse($entity->canEditMetadata($metadata, 27));
+	}
+
+	/**
+	 * @group UserCapabilities
+	 */
+	public function testCanEditMetadataWhenCanEdit() {
+		$this->setupEntity(29, 'user', '');
+		$entity = $this->setupEntity(30, 'object', 'foo', [
+			'owner_guid' => 29,
+		]);
+
+		$this->setupServices();
+		$this->assertTrue($entity->canEdit(29));
+		$this->assertTrue($entity->canEditMetadata(null, 29));
+	}
+
+	/**
+	 * @group UserCapabilities
+	 */
+	public function testCanOverrideMetadataPermissionsWithAHook() {
+
+		$this->setupEntity(31, 'user', '');
+		$entity = $this->setupEntity(32, 'object', 'foo', [
+			'owner_guid' => 31,
+		]);
+		$metadata = new \ElggMetadata((object) [
+					'owner_guid' => 31,
+					'entity_guid' => $entity->guid,
+		]);
+		$this->setupServices();
+		$this->assertTrue($entity->canEditMetadata($metadata, 31));
+
+		$this->hooks->registerHandler('permissions_check:metadata', 'object', function($hook, $type, $return, $params) {
+			$this->assertInstanceOf(\ElggEntity::class, $params['entity']);
+			$this->assertInstanceOf(\ElggUser::class, $params['user']);
+			$this->assertInstanceOf(\ElggMetadata::class, $params['metadata']);
+			$this->assertEquals('object', $type);
+			$this->assertTrue($return);
+			return false;
+		});
+		$this->setupServices();
+		$this->assertFalse($entity->canEditMetadata($metadata, 31));
+	}
+
+	/**
+	 * @group UserCapabilities
+	 */
+	public function testDefaultCanCommentPermissions() {
+
+		$user = $this->setupEntity(33, 'user', '');
+		$object = $this->setupEntity(34, 'object', 'foo', [
+			'owner_guid' => 33,
+		]);
+		$group = $this->setupEntity(35, 'group', '', [
+			'owner_guid' => 33,
+		]);
+		$entity = $this->getMockBuilder(\ElggEntity::class)
+				->setMethods(['__get', 'getDisplayName', 'setDisplayName']) // keep origin canComment method
+				->disableOriginalConstructor()
+				->getMock();
+		$entity->expects($this->any())
+				->method('__get')
+				->will($this->returnValueMap([
+							['guid', 36],
+							['owner_guid', 33]
+		]));
+
+		$this->setupServices();
+		$this->assertFalse($user->canComment(33));
+		$this->assertTrue($object->canComment(33));
+		$this->assertFalse($group->canComment(33));
+		$this->assertNull($entity->canComment(33));
```

Comment:
I don't know. That's what it is design. The value is then checked by extending classes. I am planning to tackle this wihen I work on commenting permissions.